### ISSUE TITLE
Bits for base address 0x41580, frame 31 

### ISSUE THREAD

**Issue by tmichalak** [SymbiFlow/prjxray#960]
Following 167 bits reported as missing for Litex:
<pre>
# In frame 0x0040159f 167 bits were not converted.
{ unknown_bit = "0040159f_96_0", unknown_segment = "0x00401580", unknown_segbit = "31_3072" }
{ unknown_bit = "0040159f_0_19", unknown_segment = "0x00401580", unknown_segbit = "31_19" }
{ unknown_bit = "0040159f_32_19", unknown_segment = "0x00401580", unknown_segbit = "31_1043" }
{ unknown_bit = "0040159f_0_25", unknown_segment = "0x00401580", unknown_segbit = "31_25" }
{ unknown_bit = "0040159f_0_26", unknown_segment = "0x00401580", unknown_segbit = "31_26" }
{ unknown_bit = "0040159f_32_25", unknown_segment = "0x00401580", unknown_segbit = "31_1049" }
{ unknown_bit = "0040159f_32_26", unknown_segment = "0x00401580", unknown_segbit = "31_1050" }
{ unknown_bit = "0040159f_1_1", unknown_segment = "0x00401580", unknown_segbit = "31_33" }
{ unknown_bit = "0040159f_33_1", unknown_segment = "0x00401580", unknown_segbit = "31_1057" }
{ unknown_bit = "0040159f_1_15", unknown_segment = "0x00401580", unknown_segbit = "31_47" }
{ unknown_bit = "0040159f_65_19", unknown_segment = "0x00401580", unknown_segbit = "31_2099" }
{ unknown_bit = "0040159f_97_19", unknown_segment = "0x00401580", unknown_segbit = "31_3123" }
{ unknown_bit = "0040159f_1_23", unknown_segment = "0x00401580", unknown_segbit = "31_55" }
{ unknown_bit = "0040159f_65_25", unknown_segment = "0x00401580", unknown_segbit = "31_2105" }
{ unknown_bit = "0040159f_65_26", unknown_segment = "0x00401580", unknown_segbit = "31_2106" }
{ unknown_bit = "0040159f_97_25", unknown_segment = "0x00401580", unknown_segbit = "31_3129" }
{ unknown_bit = "0040159f_97_26", unknown_segment = "0x00401580", unknown_segbit = "31_3130" }
{ unknown_bit = "0040159f_65_31", unknown_segment = "0x00401580", unknown_segbit = "31_2111" }
{ unknown_bit = "0040159f_98_1", unknown_segment = "0x00401580", unknown_segbit = "31_3137" }
{ unknown_bit = "0040159f_66_6", unknown_segment = "0x00401580", unknown_segbit = "31_2118" }
{ unknown_bit = "0040159f_66_14", unknown_segment = "0x00401580", unknown_segbit = "31_2126" }
{ unknown_bit = "0040159f_66_15", unknown_segment = "0x00401580", unknown_segbit = "31_2127" }
{ unknown_bit = "0040159f_66_18", unknown_segment = "0x00401580", unknown_segbit = "31_2130" }
{ unknown_bit = "0040159f_2_28", unknown_segment = "0x00401580", unknown_segbit = "31_92" }
{ unknown_bit = "0040159f_34_28", unknown_segment = "0x00401580", unknown_segbit = "31_1116" }
{ unknown_bit = "0040159f_66_29", unknown_segment = "0x00401580", unknown_segbit = "31_2141" }
{ unknown_bit = "0040159f_3_0", unknown_segment = "0x00401580", unknown_segbit = "31_96" }
{ unknown_bit = "0040159f_35_0", unknown_segment = "0x00401580", unknown_segbit = "31_1120" }
{ unknown_bit = "0040159f_99_9", unknown_segment = "0x00401580", unknown_segbit = "31_3177" }
{ unknown_bit = "0040159f_67_28", unknown_segment = "0x00401580", unknown_segbit = "31_2172" }
{ unknown_bit = "0040159f_99_28", unknown_segment = "0x00401580", unknown_segbit = "31_3196" }
{ unknown_bit = "0040159f_68_0", unknown_segment = "0x00401580", unknown_segbit = "31_2176" }
{ unknown_bit = "0040159f_100_0", unknown_segment = "0x00401580", unknown_segbit = "31_3200" }
{ unknown_bit = "0040159f_4_19", unknown_segment = "0x00401580", unknown_segbit = "31_147" }
{ unknown_bit = "0040159f_36_19", unknown_segment = "0x00401580", unknown_segbit = "31_1171" }
{ unknown_bit = "0040159f_4_25", unknown_segment = "0x00401580", unknown_segbit = "31_153" }
{ unknown_bit = "0040159f_4_26", unknown_segment = "0x00401580", unknown_segbit = "31_154" }
{ unknown_bit = "0040159f_36_25", unknown_segment = "0x00401580", unknown_segbit = "31_1177" }
{ unknown_bit = "0040159f_36_26", unknown_segment = "0x00401580", unknown_segbit = "31_1178" }
{ unknown_bit = "0040159f_5_1", unknown_segment = "0x00401580", unknown_segbit = "31_161" }
{ unknown_bit = "0040159f_37_1", unknown_segment = "0x00401580", unknown_segbit = "31_1185" }
{ unknown_bit = "0040159f_69_19", unknown_segment = "0x00401580", unknown_segbit = "31_2227" }
{ unknown_bit = "0040159f_69_25", unknown_segment = "0x00401580", unknown_segbit = "31_2233" }
{ unknown_bit = "0040159f_69_26", unknown_segment = "0x00401580", unknown_segbit = "31_2234" }
{ unknown_bit = "0040159f_70_1", unknown_segment = "0x00401580", unknown_segbit = "31_2241" }
{ unknown_bit = "0040159f_38_3", unknown_segment = "0x00401580", unknown_segbit = "31_1219" }
{ unknown_bit = "0040159f_6_28", unknown_segment = "0x00401580", unknown_segbit = "31_220" }
{ unknown_bit = "0040159f_38_28", unknown_segment = "0x00401580", unknown_segbit = "31_1244" }
{ unknown_bit = "0040159f_7_0", unknown_segment = "0x00401580", unknown_segbit = "31_224" }
{ unknown_bit = "0040159f_39_0", unknown_segment = "0x00401580", unknown_segbit = "31_1248" }
{ unknown_bit = "0040159f_71_28", unknown_segment = "0x00401580", unknown_segbit = "31_2300" }
{ unknown_bit = "0040159f_72_0", unknown_segment = "0x00401580", unknown_segbit = "31_2304" }
{ unknown_bit = "0040159f_8_19", unknown_segment = "0x00401580", unknown_segbit = "31_275" }
{ unknown_bit = "0040159f_40_19", unknown_segment = "0x00401580", unknown_segbit = "31_1299" }
{ unknown_bit = "0040159f_8_25", unknown_segment = "0x00401580", unknown_segbit = "31_281" }
{ unknown_bit = "0040159f_8_26", unknown_segment = "0x00401580", unknown_segbit = "31_282" }
{ unknown_bit = "0040159f_40_25", unknown_segment = "0x00401580", unknown_segbit = "31_1305" }
{ unknown_bit = "0040159f_40_26", unknown_segment = "0x00401580", unknown_segbit = "31_1306" }
{ unknown_bit = "0040159f_9_1", unknown_segment = "0x00401580", unknown_segbit = "31_289" }
{ unknown_bit = "0040159f_41_1", unknown_segment = "0x00401580", unknown_segbit = "31_1313" }
{ unknown_bit = "0040159f_41_15", unknown_segment = "0x00401580", unknown_segbit = "31_1327" }
{ unknown_bit = "0040159f_73_19", unknown_segment = "0x00401580", unknown_segbit = "31_2355" }
{ unknown_bit = "0040159f_41_23", unknown_segment = "0x00401580", unknown_segbit = "31_1335" }
{ unknown_bit = "0040159f_73_25", unknown_segment = "0x00401580", unknown_segbit = "31_2361" }
{ unknown_bit = "0040159f_73_26", unknown_segment = "0x00401580", unknown_segbit = "31_2362" }
{ unknown_bit = "0040159f_74_1", unknown_segment = "0x00401580", unknown_segbit = "31_2369" }
{ unknown_bit = "0040159f_10_28", unknown_segment = "0x00401580", unknown_segbit = "31_348" }
{ unknown_bit = "0040159f_42_28", unknown_segment = "0x00401580", unknown_segbit = "31_1372" }
{ unknown_bit = "0040159f_11_0", unknown_segment = "0x00401580", unknown_segbit = "31_352" }
{ unknown_bit = "0040159f_43_0", unknown_segment = "0x00401580", unknown_segbit = "31_1376" }
{ unknown_bit = "0040159f_12_19", unknown_segment = "0x00401580", unknown_segbit = "31_403" }
{ unknown_bit = "0040159f_44_19", unknown_segment = "0x00401580", unknown_segbit = "31_1427" }
{ unknown_bit = "0040159f_12_25", unknown_segment = "0x00401580", unknown_segbit = "31_409" }
{ unknown_bit = "0040159f_12_26", unknown_segment = "0x00401580", unknown_segbit = "31_410" }
{ unknown_bit = "0040159f_44_25", unknown_segment = "0x00401580", unknown_segbit = "31_1433" }
{ unknown_bit = "0040159f_44_26", unknown_segment = "0x00401580", unknown_segbit = "31_1434" }
{ unknown_bit = "0040159f_13_1", unknown_segment = "0x00401580", unknown_segbit = "31_417" }
{ unknown_bit = "0040159f_45_1", unknown_segment = "0x00401580", unknown_segbit = "31_1441" }
{ unknown_bit = "0040159f_77_11", unknown_segment = "0x00401580", unknown_segbit = "31_2475" }
{ unknown_bit = "0040159f_77_17", unknown_segment = "0x00401580", unknown_segbit = "31_2481" }
{ unknown_bit = "0040159f_77_19", unknown_segment = "0x00401580", unknown_segbit = "31_2483" }
{ unknown_bit = "0040159f_77_25", unknown_segment = "0x00401580", unknown_segbit = "31_2489" }
{ unknown_bit = "0040159f_77_26", unknown_segment = "0x00401580", unknown_segbit = "31_2490" }
{ unknown_bit = "0040159f_78_1", unknown_segment = "0x00401580", unknown_segbit = "31_2497" }
{ unknown_bit = "0040159f_78_15", unknown_segment = "0x00401580", unknown_segbit = "31_2511" }
{ unknown_bit = "0040159f_46_28", unknown_segment = "0x00401580", unknown_segbit = "31_1500" }
{ unknown_bit = "0040159f_78_29", unknown_segment = "0x00401580", unknown_segbit = "31_2525" }
{ unknown_bit = "0040159f_47_0", unknown_segment = "0x00401580", unknown_segbit = "31_1504" }
{ unknown_bit = "0040159f_79_28", unknown_segment = "0x00401580", unknown_segbit = "31_2556" }
{ unknown_bit = "0040159f_80_0", unknown_segment = "0x00401580", unknown_segbit = "31_2560" }
{ unknown_bit = "0040159f_16_19", unknown_segment = "0x00401580", unknown_segbit = "31_531" }
{ unknown_bit = "0040159f_48_19", unknown_segment = "0x00401580", unknown_segbit = "31_1555" }
{ unknown_bit = "0040159f_16_25", unknown_segment = "0x00401580", unknown_segbit = "31_537" }
{ unknown_bit = "0040159f_16_26", unknown_segment = "0x00401580", unknown_segbit = "31_538" }
{ unknown_bit = "0040159f_48_25", unknown_segment = "0x00401580", unknown_segbit = "31_1561" }
{ unknown_bit = "0040159f_48_26", unknown_segment = "0x00401580", unknown_segbit = "31_1562" }
{ unknown_bit = "0040159f_17_1", unknown_segment = "0x00401580", unknown_segbit = "31_545" }
{ unknown_bit = "0040159f_49_1", unknown_segment = "0x00401580", unknown_segbit = "31_1569" }
{ unknown_bit = "0040159f_17_10", unknown_segment = "0x00401580", unknown_segbit = "31_554" }
{ unknown_bit = "0040159f_17_15", unknown_segment = "0x00401580", unknown_segbit = "31_559" }
{ unknown_bit = "0040159f_81_19", unknown_segment = "0x00401580", unknown_segbit = "31_2611" }
{ unknown_bit = "0040159f_17_22", unknown_segment = "0x00401580", unknown_segbit = "31_566" }
{ unknown_bit = "0040159f_17_23", unknown_segment = "0x00401580", unknown_segbit = "31_567" }
{ unknown_bit = "0040159f_81_25", unknown_segment = "0x00401580", unknown_segbit = "31_2617" }
{ unknown_bit = "0040159f_81_26", unknown_segment = "0x00401580", unknown_segbit = "31_2618" }
{ unknown_bit = "0040159f_82_1", unknown_segment = "0x00401580", unknown_segbit = "31_2625" }
{ unknown_bit = "0040159f_18_28", unknown_segment = "0x00401580", unknown_segbit = "31_604" }
{ unknown_bit = "0040159f_19_0", unknown_segment = "0x00401580", unknown_segbit = "31_608" }
{ unknown_bit = "0040159f_83_28", unknown_segment = "0x00401580", unknown_segbit = "31_2684" }
{ unknown_bit = "0040159f_84_0", unknown_segment = "0x00401580", unknown_segbit = "31_2688" }
{ unknown_bit = "0040159f_20_19", unknown_segment = "0x00401580", unknown_segbit = "31_659" }
{ unknown_bit = "0040159f_20_25", unknown_segment = "0x00401580", unknown_segbit = "31_665" }
{ unknown_bit = "0040159f_20_26", unknown_segment = "0x00401580", unknown_segbit = "31_666" }
{ unknown_bit = "0040159f_21_1", unknown_segment = "0x00401580", unknown_segbit = "31_673" }
{ unknown_bit = "0040159f_53_19", unknown_segment = "0x00401580", unknown_segbit = "31_1715" }
{ unknown_bit = "0040159f_85_19", unknown_segment = "0x00401580", unknown_segbit = "31_2739" }
{ unknown_bit = "0040159f_53_25", unknown_segment = "0x00401580", unknown_segbit = "31_1721" }
{ unknown_bit = "0040159f_53_26", unknown_segment = "0x00401580", unknown_segbit = "31_1722" }
{ unknown_bit = "0040159f_85_25", unknown_segment = "0x00401580", unknown_segbit = "31_2745" }
{ unknown_bit = "0040159f_85_26", unknown_segment = "0x00401580", unknown_segbit = "31_2746" }
{ unknown_bit = "0040159f_54_1", unknown_segment = "0x00401580", unknown_segbit = "31_1729" }
{ unknown_bit = "0040159f_86_1", unknown_segment = "0x00401580", unknown_segbit = "31_2753" }
{ unknown_bit = "0040159f_22_28", unknown_segment = "0x00401580", unknown_segbit = "31_732" }
{ unknown_bit = "0040159f_23_0", unknown_segment = "0x00401580", unknown_segbit = "31_736" }
{ unknown_bit = "0040159f_55_28", unknown_segment = "0x00401580", unknown_segbit = "31_1788" }
{ unknown_bit = "0040159f_56_0", unknown_segment = "0x00401580", unknown_segbit = "31_1792" }
{ unknown_bit = "0040159f_24_19", unknown_segment = "0x00401580", unknown_segbit = "31_787" }
{ unknown_bit = "0040159f_24_25", unknown_segment = "0x00401580", unknown_segbit = "31_793" }
{ unknown_bit = "0040159f_24_26", unknown_segment = "0x00401580", unknown_segbit = "31_794" }
{ unknown_bit = "0040159f_25_1", unknown_segment = "0x00401580", unknown_segbit = "31_801" }
{ unknown_bit = "0040159f_89_19", unknown_segment = "0x00401580", unknown_segbit = "31_2867" }
{ unknown_bit = "0040159f_57_19", unknown_segment = "0x00401580", unknown_segbit = "31_1843" }
{ unknown_bit = "0040159f_89_25", unknown_segment = "0x00401580", unknown_segbit = "31_2873" }
{ unknown_bit = "0040159f_89_26", unknown_segment = "0x00401580", unknown_segbit = "31_2874" }
{ unknown_bit = "0040159f_57_25", unknown_segment = "0x00401580", unknown_segbit = "31_1849" }
{ unknown_bit = "0040159f_57_26", unknown_segment = "0x00401580", unknown_segbit = "31_1850" }
{ unknown_bit = "0040159f_89_31", unknown_segment = "0x00401580", unknown_segbit = "31_2879" }
{ unknown_bit = "0040159f_58_1", unknown_segment = "0x00401580", unknown_segbit = "31_1857" }
{ unknown_bit = "0040159f_26_3", unknown_segment = "0x00401580", unknown_segbit = "31_835" }
{ unknown_bit = "0040159f_90_6", unknown_segment = "0x00401580", unknown_segbit = "31_2886" }
{ unknown_bit = "0040159f_90_14", unknown_segment = "0x00401580", unknown_segbit = "31_2894" }
{ unknown_bit = "0040159f_90_15", unknown_segment = "0x00401580", unknown_segbit = "31_2895" }
{ unknown_bit = "0040159f_90_18", unknown_segment = "0x00401580", unknown_segbit = "31_2898" }
{ unknown_bit = "0040159f_26_28", unknown_segment = "0x00401580", unknown_segbit = "31_860" }
{ unknown_bit = "0040159f_90_29", unknown_segment = "0x00401580", unknown_segbit = "31_2909" }
{ unknown_bit = "0040159f_27_0", unknown_segment = "0x00401580", unknown_segbit = "31_864" }
{ unknown_bit = "0040159f_59_28", unknown_segment = "0x00401580", unknown_segbit = "31_1916" }
{ unknown_bit = "0040159f_91_28", unknown_segment = "0x00401580", unknown_segbit = "31_2940" }
{ unknown_bit = "0040159f_60_0", unknown_segment = "0x00401580", unknown_segbit = "31_1920" }
{ unknown_bit = "0040159f_92_0", unknown_segment = "0x00401580", unknown_segbit = "31_2944" }
{ unknown_bit = "0040159f_28_19", unknown_segment = "0x00401580", unknown_segbit = "31_915" }
{ unknown_bit = "0040159f_28_25", unknown_segment = "0x00401580", unknown_segbit = "31_921" }
{ unknown_bit = "0040159f_28_26", unknown_segment = "0x00401580", unknown_segbit = "31_922" }
{ unknown_bit = "0040159f_29_1", unknown_segment = "0x00401580", unknown_segbit = "31_929" }
{ unknown_bit = "0040159f_29_15", unknown_segment = "0x00401580", unknown_segbit = "31_943" }
{ unknown_bit = "0040159f_93_19", unknown_segment = "0x00401580", unknown_segbit = "31_2995" }
{ unknown_bit = "0040159f_61_19", unknown_segment = "0x00401580", unknown_segbit = "31_1971" }
{ unknown_bit = "0040159f_29_23", unknown_segment = "0x00401580", unknown_segbit = "31_951" }
{ unknown_bit = "0040159f_93_25", unknown_segment = "0x00401580", unknown_segbit = "31_3001" }
{ unknown_bit = "0040159f_93_26", unknown_segment = "0x00401580", unknown_segbit = "31_3002" }
{ unknown_bit = "0040159f_61_25", unknown_segment = "0x00401580", unknown_segbit = "31_1977" }
{ unknown_bit = "0040159f_61_26", unknown_segment = "0x00401580", unknown_segbit = "31_1978" }
{ unknown_bit = "0040159f_62_1", unknown_segment = "0x00401580", unknown_segbit = "31_1985" }
{ unknown_bit = "0040159f_94_1", unknown_segment = "0x00401580", unknown_segbit = "31_3009" }
{ unknown_bit = "0040159f_30_28", unknown_segment = "0x00401580", unknown_segbit = "31_988" }
{ unknown_bit = "0040159f_31_0", unknown_segment = "0x00401580", unknown_segbit = "31_992" }
{ unknown_bit = "0040159f_95_28", unknown_segment = "0x00401580", unknown_segbit = "31_3068" }
</pre>
are solved by the following bits:
<pre>
LIOB33.IOB_Y1.31_28 31_28
LIOB33.IOB_Y1.31_32 31_32
LIOB33.IOB_Y1.31_83 31_83
LIOB33.IOB_Y1.31_89 31_89
LIOB33.IOB_Y1.31_90 31_90
LIOB33.IOB_Y1.31_95 31_95
LIOB33.IOB_Y1.31_97 31_97
LIOB33.IOB_Y1.31_102 31_102
LIOB33.IOB_Y1.31_110 31_110
LIOB33.IOB_Y1.31_111 31_111
LIOB33.IOB_Y1.31_114 31_114
LIOB33.IOB_Y1.31_119 31_119
LIOB33.IOB_Y1.31_125 31_125
</pre>

The seem to be OLOGIC/ILOGIC related

**Comment by acomodi** [comment]
I have checked and `037-ioi-pips` fuzzer solves the following:
```
31_111
31_119
31_125
```

**Comment by litghost** [comment]
```
LIOB33.IOB_Y1.31_90 31_90
LIOB33.IOB_Y1.31_102 31_102
LIOB33.IOB_Y1.31_110 31_110
LIOB33.IOB_Y1.31_114 31_114
```

are solved by #971


Remaining bits:
```
LIOB33.IOB_Y1.31_28 31_28
LIOB33.IOB_Y1.31_32 31_32
LIOB33.IOB_Y1.31_83 31_83
LIOB33.IOB_Y1.31_89 31_89
LIOB33.IOB_Y1.31_95 31_95
LIOB33.IOB_Y1.31_97 31_97
```

**Comment by acomodi** [comment]
`LIOB33.IOB_Y1.31_95 31_95` is most probably related to the differential output mode.
